### PR TITLE
Upgrade all third-party libraries

### PR DESCRIPTION
**Pull request by sherman-the-tank**
This is only intended for 5.0

### CODE REVIEW

**Review by codesigner — APPROVED** [review]

File: project/patches/breakpad-2021-11-11.patch [hunk 1]
```diff
@@ -1,13 +0,0 @@
-diff -ur a/src/client/linux/handler/exception_handler.cc b/src/client/linux/handler/exception_handler.cc
---- a/src/client/linux/handler/exception_handler.cc     2021-11-11 10:57:53.465471527 -0800
-+++ b/src/client/linux/handler/exception_handler.cc     2021-11-11 11:00:01.639701219 -0800
-@@ -138,7 +138,7 @@
-   // SIGSTKSZ may be too small to prevent the signal handlers from overrunning
-   // the alternative stack. Ensure that the size of the alternative stack is
-   // large enough.
--  static const unsigned kSigStackSize = std::max(16384, SIGSTKSZ);
-+  static const unsigned kSigStackSize = std::max(static_cast<decltype(SIGSTKSZ)>(16384), SIGSTKSZ);
```
Comment (codesigner) [inline project/patches/breakpad-2021-11-11.patch:9]: the latest version is same as before, need check if this patch need in the new version

File: project/patches/fbthrift-2021-11-29.patch [hunk 2]
```diff
@@ -1,24 +0,0 @@
-diff -ur a/thrift/lib/cpp2/async/HeaderServerChannel.h b/thrift/lib/cpp2/async/HeaderServerChannel.h
---- a/thrift/lib/cpp2/async/HeaderServerChannel.h	2021-11-29 14:14:12.000000000 +0800
-+++ b/thrift/lib/cpp2/async/HeaderServerChannel.h	2022-10-13 16:22:03.142090462 +0800
-@@ -108,7 +108,7 @@
-     }
- 
-     bool isOneway() const override {
--      return header_->getSequenceNumber() == ONEWAY_REQUEST_ID;
-+      return header_ && header_->getSequenceNumber() == ONEWAY_REQUEST_ID;
```
Comment (codesigner) [inline project/patches/fbthrift-2021-11-29.patch:9]: this may have some issue before, need check whether the new version need this patch

**Review by codesigner — COMMENTED** [review]
some deleted patch need a double check, if the new version fixed the issue , then we can delete it , otherwise we need a new patch for the new version.

**Comment by cangfengzhs** [comment]
Will we use absl later? I haven't used it, but according to the information I have, it and Folly are both basic libraries of C++ and many functions are repetitive?

**Pull request merged**